### ISSUE TITLE
citre-peek-through does not work with eglot backend

### ISSUE THREAD

**Issue by mplanchard** [universal-ctags/citre#177]
## Describe the bug

When using citre-peek with the eglot backend, the initial `peek` works well, but subsequently using `citre-peek-through` results in the following:

```
citre-peek--get-tags: Can’t find definition: "IntegrationInfo" (eglot backend), "IntegrationInfo" (tags backend), "IntegrationInfo" (global backend).
```

Backtrace in the fold:
<details>
<summary>Backtrace</summary>
<pre>
Debugger entered--Lisp error: (file-missing "Searching for program" "No such file or directory" "rust-analyzer")
  make-process(:name "EGLOT (spec-protect/(rustic-mode))" :command ("rust-analyzer") :connection-type pipe :coding utf-8-emacs-unix :noquery t :stderr #<buffer *EGLOT (spec-protect/(rustic-mode)) stderr*> :file-handler t)
  #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)()
  #f(compiled-function (conn slots) #<bytecode 0xc0b47a118db30dd>)(#<eglot-rust-analyzer eglot-rust-analyzer-1b56f21> (:name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)))
  apply(#f(compiled-function (conn slots) #<bytecode 0xc0b47a118db30dd>) (#<eglot-rust-analyzer eglot-rust-analyzer-1b56f21> (:name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>))))
  #f(compiled-function (&rest args) #<bytecode 0x5553a698680a115>)(#<eglot-rust-analyzer eglot-rust-analyzer-1b56f21> (:name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)))
  apply(#f(compiled-function (&rest args) #<bytecode 0x5553a698680a115>) #<eglot-rust-analyzer eglot-rust-analyzer-1b56f21> (:name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)))
  initialize-instance(#<eglot-rust-analyzer eglot-rust-analyzer-1b56f21> (:name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)))
  #f(compiled-function (class &rest slots) "Default constructor for CLASS `eieio-default-superclass'.\nSLOTS are the initialization slots used by `initialize-instance'.\nThis static method is called when an object is constructed.\nIt allocates the vector used to represent an EIEIO object, and then\ncalls `initialize-instance' on that object." #<bytecode -0x168b5f529242c257>)(eglot-rust-analyzer :name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>))
  apply(#f(compiled-function (class &rest slots) "Default constructor for CLASS `eieio-default-superclass'.\nSLOTS are the initialization slots used by `initialize-instance'.\nThis static method is called when an object is constructed.\nIt allocates the vector used to represent an EIEIO object, and then\ncalls `initialize-instance' on that object." #<bytecode -0x168b5f529242c257>) eglot-rust-analyzer (:name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)))
  make-instance(eglot-rust-analyzer :name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown :process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>))
  apply(make-instance eglot-rust-analyzer :name "EGLOT (spec-protect/(rustic-mode))" :events-buffer-config (:size 2000000 :format full) :notification-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076894187b8aa>) :request-dispatcher #f(compiled-function (server method params) #<bytecode 0x1f1076895f87b8aa>) :on-shutdown eglot--on-shutdown (:process #f(compiled-function () #<bytecode 0x12092fa9ac5a12aa>)))
  eglot--connect((rustic-mode) (vc Git "~/s/spec-protect/") eglot-rust-analyzer ("rust-analyzer") ("rust"))
  apply(eglot--connect ((rustic-mode) (vc Git "~/s/spec-protect/") eglot-rust-analyzer ("rust-analyzer") ("rust")))
  #f(compiled-function () #<bytecode 0xa451b1e1db37ba5>)()
  citre--eglot-after-jump()
  apply(citre--eglot-after-jump nil)
  citre--call-backend-fn(eglot after-jump-fn)
  citre-backend-after-jump(eglot)
  citre-peek-through()
  funcall-interactively(citre-peek-through)
  command-execute(citre-peek-through)
</pre>
</details>

## To Reproduce

Steps to reproduce:

Ensure you have `rust-analyzer` installed, and then run  `$ emacs -Q`

Evaluate the following in the scratch buffer:

```elisp
(add-to-list 'package-archives
             '("melpa-stable" . "https://stable.melpa.org/packages/") t)

(use-package citre :ensure t)
(use-package rust-mode :ensure t)

;; only necessary for me because the rust project is a nix project w/an isolated rust-analyzer, not
;; necessary if you have it installed globally
(add-to-list 'exec-path "/nix/store/m8db0w1l880a5i8lxymjf34i7xd6viyg-rust-analyzer-2024-04-29/bin/")
```

Navigate to any rust file, and run `citre-peek` on an identifier. From there, run `citre-peek-through` and select a target. Any target results in the error above.

## System information

- OS: NixOS 24.05
- Emacs version: 31.0.50
- Are you using the latest version of Citre on master branch: no, using the version on melpa stable

## Additional context

n/a

**Comment by AmaiKinono** [comment]
Hi, and sorry for the delay.

I got some chance to play with rust. From what I see, rust analyzer spent some time building the cache, then all the `xref` and Citre commands works fine:

![image](https://github.com/user-attachments/assets/3c02e4a3-58f3-4310-b8b7-6bf3612bdaf3)

My guess is you didn't run `M-x eglot` before using Citre commands, or the building process isn't finished.

**Comment by mplanchard** [comment]
Oh no worries, I had forgotten about opening this, and I appreciate your looking into it. I resolved the issue some time ago, although I do not remember exactly what I did that fixed it. 

It looks from my backtrace like that there may have been an issue in the peek window context of finding rust-analyzer, which in my environment is generally defined by a nix-shell activated by dotenv. It is quite possible that tidying up my `envrc` setup is what resolved the problem.

Thank you again for the checking into it!
